modules/core/arrow-annotations-processor/src/main/java/arrow/common/utils/ProcessorUtils.kt

```diff
@@ -13,121 +13,151 @@ import javax.lang.model.element.VariableElement
 
 interface ProcessorUtils : KotlinMetadataUtils {
 
-  fun KotlinMetadata.asClassOrPackageDataWrapper(classElement: TypeElement): ClassOrPackageDataWrapper? {
-    val `package` = elementUtils.getPackageOf(classElement).toString()
-    return when (this) {
-      is KotlinClassMetadata -> data.asClassOrPackageDataWrapper(`package`)
-      is KotlinPackageMetadata -> data.asClassOrPackageDataWrapper(`package`)
-      else -> null
+    fun getConstructorParamNames(element: Element): List<String> = element.kotlinMetadata
```

Comment:
Better as an extfun or extension field actually :D Same for the others in this file.